modules/core/site.js

```diff
@@ -528,11 +528,21 @@ function Message_List() {
         var aval;
         var bval;
         var sort_result = listitems.sort(function(a, b) {
-            const sortField = fld.replace('-', '');
-            if (['arrival', 'date'].includes(sortField)) {
-                aval = new Date($(`input.${sortField}`, $('td.dates', a)).val());
-                bval = new Date($(`input.${sortField}`, $('td.dates', b)).val());
-                if (fld.startsWith('-')) {
+            switch (Math.abs(fld)) {
+                case 1:
+                case 2:
+                case 3:
+                    aval = $($('td', a)[Math.abs(fld)]).text().replace(/^\s+/g, '');
+                    bval = $($('td', b)[Math.abs(fld)]).text().replace(/^\s+/g, '');
+                    break;
+                case 4:
+                default:
+                    aval = $('input', $($('td', a)[Math.abs(fld)])).val();
+                    bval = $('input', $($('td', b)[Math.abs(fld)])).val();
+                    break;
+            }
+            if (fld == 4 || fld == -4 || !fld) {
+                if (fld == -4) {
```

Comment:
This rebase was not accurate. See the actual state in this [diff](https://github.com/cypht-org/cypht/commit/dedcfb7dfb571a1db79db839e4a787cc9a082680#diff-eef52370546e86abe4041840d8277b2a1fb9d3716ffc670451b4b839884bfa2e).

Reply:
You are still omitting important parts of the code while including unnecessary ones. Please refer to the version on master to pick the correct code.